models/ResourceService.cfc

```diff
@@ -26,28 +30,34 @@ component singleton accessors="true" {
 	 * @controller.inject coldbox
 	 * @i18n cbi18n module
 	 * @i18n.inject i18n@cbi18n
+	 * @settings cbi18n settings
+	 * @settings.inject coldbox:moduleSettings:cbi18n
+	 * 
+	 * @throws cbi18n.InvalidConfiguration
 	 */
-	function init( required controller, required i18n ) {
+	function init( required controller, required i18n, required settings ) {
```

Comment:
I am wondering why did you move the settinghs to the constructor and not to the onDiComplete method?  Because now you have double injection going on, as properties and as constructors?

Reply:
I agree. It should be one or the other, not both. I was confused because I saw the constructor method in another module. But onDiComplete is cleaner, as lang as we are sure it will always be instantiated with Wirebox.
In qb default settings are injected in the object by creating some initArg stuff in a binder.map. But of course we can do the same with .property() or .setter(). I'll change it to onDiComplete again

Reply:
Ok, controller and i18n were already done in the constructor. I moved everything to property injections, created onDiComplete again.  Since we have a .settings property it didn't make sense to populate all these individual properties again from settings. So init and onDiComplete is a lot simpler now. (I think this was all left from the original instance.something style.
And modified the tests a little.